src/App.tsx

```diff
@@ -15,48 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getSortedGoods(goodsArr: string[], sortField: SortType): string[] {
+  if (sortField !== SortType.Default) {
+    goodsArr.sort((a, b): number => {
+      switch (sortField) {
+        case SortType.Alphabet: {
+          return a.localeCompare(b);
+        }
+
+        case SortType.Length: {
+          return a.length - b.length;
+        }
+
+        default: {
+          return 0;
+        }
+      }
+    });
+  }
+
+  return goodsArr;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortField, setSortField] = useState(SortType.Default);
+  const isHideResetBtn = sortField || isReversed;
+
+  const resetField = () => {
+    setIsReversed(false);
+    setSortField(SortType.Default);
+  };
+
+  const goods = getSortedGoods([...goodsFromServer], sortField);
```

Comment:
It is better to transfer the creation of a copy of the array to the middle of the `getSortedGoods` function.